src/FluentResults/Factories/ResultHelper.cs

```diff
@@ -41,7 +41,9 @@ public static Result<IEnumerable<TValue>> MergeWithValue<TValue>(IEnumerable<Res
 
         public static bool HasException<TException>(List<IError> errors, Func<TException, bool> predicate) where TException : Exception
         {
-            var anyErrors = errors.Any(e => e.Reasons.OfType<IExceptionalError>().Any(r => r.Exception is TException exceptionOfTException && predicate(exceptionOfTException)));
+            var anyErrors = errors.Any(e => (e is ExceptionalError && ((ExceptionalError)e).Exception is TException rootExceptionOfTException && predicate(rootExceptionOfTException)) ||
+                                             e.Reasons.OfType<ExceptionalError>().Any(r => r.Exception is TException exceptionOfTException && predicate(exceptionOfTException)));
```

Comment:
I don't understand why this second line is needed. I merged into the master and removed the line and all tests are green.